test/gtest/gpu_nchw_nhwc_transpose.cpp

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "gpu_nchw_nhwc_transpose.hpp"
+
+DEFINE_DEFAULT_TO_NHWC_TEST(float);
+DEFINE_DEFAULT_TO_NHWC_TEST(uint16_t);
+DEFINE_DEFAULT_TO_NHWC_TEST(uint8_t);
+
+DEFINE_NHWC_TO_DEFAULT_TEST(float);
+DEFINE_NHWC_TO_DEFAULT_TEST(uint16_t);
+DEFINE_NHWC_TO_DEFAULT_TEST(uint8_t);
```

Comment:
`gpu_nchw_nhwc_transpose.hpp` is included only once. Please move macros to the end of it, remove this file and rename `gpu_nchw_nhwc_transpose.hpp` to `gpu_nchw_nhwc_transpose.cpp`.

Reply:
Hi Artem, I just realized that I misread your comment the other day. Yes, it makes sense to move all of the code into the cpp file. I'll do that.